test/modules/DirectMethodSender/Program.cs

```diff
@@ -141,5 +107,31 @@ public static bool IsTestTimeUp(DateTime testStartAt)
         {
             return (Settings.Current.TestDuration == TimeSpan.Zero) || (DateTime.UtcNow - testStartAt < Settings.Current.TestDuration);
         }
+
+        // Create reporting result depending on which endpoint is being used.
+        public static TestResultBase ConstructTestResult(DirectMethodResultType directMethodResultType, Guid batchId, long counter, string result)
+        {
+            string source = Settings.Current.ModuleId + ".send";
+            switch (directMethodResultType)
+            {
+                case DirectMethodResultType.DirectMethodTestResult:
+                    return new DirectMethodTestResult(source, DateTime.UtcNow)
+                    {
+                        TrackingId = Settings.Current.TrackingId.Expect(() => new ArgumentException("TrackingId is empty")),
```

Comment:
did you already validate it in Settings? if so, just call OrDefault() method.

Reply:
No it doesn't check at Setting. 
Only when the option of `DirectMethodTestResult` is specified that the checking is done here.